### PR TITLE
Feature/security  campaign blocks

### PR DESCRIPTION
**Pull request by kycapstick**
Updated all campaign blocks (stored in the templates/blocks folder) to pass code sniffer standards. 

will need to test the following blocks:

`events-block
imagery-block
outro-cta-block
text-1up-block
text-2up-block
text-3up-block
text-card-block
text-image-block
video-block`

### CODE REVIEW

File: templates/blocks/events-block.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+					$location   = em_get_location( $em_event->location_id );
+					$categories = ( ! is_null( $em_event ) ) ? $em_event->get_categories() : false;
+					?>
+				<a href="<?php print esc_attr( $event['event']->guid ); ?>" class="campaign__event">
+					<div class="campaign__event-image" 
+					<?php
+					if ( isset( $event_meta[0]->image_url ) && strlen( $event_meta[0]->image_url ) > 0 ) :
+						?>
+						style="background-image: url('<?php print esc_attr( $event_meta[0]->image_url ); ?>')"<?php endif; ?>>
```
Comment (dnaluz) [inline templates/blocks/events-block.php:69]: This is a url.  You should use esc_url_raw in the case of URLs

**Comment by dnaluz** [comment]
@kycapstick, not sure but should some of the blocks allow links in their content, I don't remember if they do.  Also you may need to allow some elements to have class attributes if there are any internal styles applied to content.

File: single-campaign.php [hunk 2]
```diff
@@ -1,133 +1,149 @@
...
+						<h1 class="campaign__hero-title"><?php print $post->post_title; ?></h1>
+						<div class="campaign__date-container">
+							<?php print $campaign_start_date; ?>
+							<?php if ( $campaign_end_date ) : ?>
+							- <?php print $campaign_end_date; ?>
+							<?php endif; ?>
+						</div>
+						<?php if ( ( $campaign_hero_cta && $mailchimp && isset( $mailchimp->id ) ) || is_preview() ) : ?>
+							<?php if ( $logged_in ) : ?>
+								<a href="<?php print ( $campaign_hero_cta_link ) ? $campaign_hero_cta_link : '#'; ?>" class="campaign__hero-cta
```
[the remark this answers is not in the document]
Comment (kycapstick) [reply single-campaign.php:50]: Sorry about the confusion here! The fixes to single-campaign and pages-campaign were still on a separate branch. I was only updating single-campaign to change the file names for the individual blocks.

File: templates/blocks/events-block.php [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+							</svg>
+							<?php else : ?>
+							<svg width="16" height="14" viewBox="0 0 16 14" fill="none" xmlns="http://www.w3.org/2000/svg">
+								<path d="M14.6666 7H11.9999L9.99992 13L5.99992 1L3.99992 7H1.33325" stroke="#737373" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
+							</svg>
+							<?php endif; ?>
+							<?php
+							print esc_html__( 'Part of ', 'community-portal' ) . ' ' . esc_html( $initiative->post_title ) . ' ';
+							if ( 'campaign' === $initiative->post_type ) {
+								print esc_html__( 'Campaign', 'community-portal' );
```
[the remark this answers is not in the document]
Comment (kycapstick) [reply templates/blocks/events-block.php:128]: It still throws an error if i don't use the esc_html__() I'm not using the e as it's being echoed as a variable but codesniffer still wants it to be escaped.

**Comment by kycapstick** [comment]
Updated most instances of wp_kses to allow for a tags and classes. I've left a tags out of the wp_kses for card descriptions as we've already communicated that since the cards are links, they can't use links within those descriptions.

**Pull request merged**